tests/src/FunctionalJavascript/Integration/AutosaveFormTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Drupal\Tests\thunder\FunctionalJavascript\Integration;
+
+use Drupal\Tests\thunder\FunctionalJavascript\ThunderFormFieldTestTrait;
+use Drupal\Tests\thunder\FunctionalJavascript\ThunderJavascriptTestBase;
+use Drupal\Tests\thunder\FunctionalJavascript\ThunderParagraphsTestTrait;
+
+/**
+ * Tests the autosave support for nodes in Thunder.
+ *
+ * @group Thunder
+ */
+class AutosaveFormTest extends ThunderJavascriptTestBase {
+
+  use ThunderFormFieldTestTrait;
+  use ThunderParagraphsTestTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Adjust the autosave form submission interval.
+    \Drupal::configFactory()
+      ->getEditable('autosave_form.settings')
+      ->set('interval', 2000)
+      ->save();
+  }
+
+  /**
+   * Tests the autosave functionality in an existing article.
+   */
+  public function testAutosaveInExistingEntity() {
+    $this->drupalGet('node/7/edit');
+    $page = $this->getSession()->getPage();
+
+    // Make some changes.
+    $this->makeFormChanges();
+
+    // Reload the page.
+    $this->drupalGet('node/7/edit');
+
+    // Reject the changes.
+    $this->pressRejectButton();
+    $this->assertEquals([5], $page->findField('field_tags[]')->getValue());
+    $this->assertEquals('Come to DrupalCon New Orleans', $page->findField('title[0][value]')->getValue());
```

Comment:
I think it's important to validate that the newly added paragraph is not there anymore.